src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -375,16 +390,17 @@ case class WorkflowActor(workflow: WorkflowDescriptor, backend: Backend)
     val completionWork = for {
       _ <- globalDataAccess.setExecutionEvents(workflow.id, callKey.scope.fullyQualifiedName, callKey.index, callKey.attempt, events)
       _ <- globalDataAccess.insertCalls(workflow.id, Seq(keyClone), backend)
-      _ = persistStatusThenAck(callKey, retryStatus, currentSender, message, callOutputs = None, returnCode = returnCode)
     } yield ()
 
-    completionWork onFailure {
-      case e =>
+    completionWork onComplete {
+      case Failure(e) =>
         logger.error(s"Preemption work failed for call ${callKey.tag}! " + e.getMessage, e)
         self ! CheckForWorkflowComplete
+      case Success(_) =>
+        self ! PersistStatus(callKey, retryStatus, None, returnCode, None, None, currentSender, message)
```

Comment:
Yep different problem but from the same bug ticket.
Because the setOutputs was failing (because of the empty string), the status was never set to "Failed", so the CallActor keep sending its `CallCompleted` message because it was never getting an ack which led to infinite retries of `setOutputs` etc...